### ISSUE TITLE
Different root URL?

### ISSUE THREAD

**Issue by screendriver** [nonpop/elm-purl#1]
`root </> s "test"` results in `/test` but in my case I need `https://foo.bar/test`.

How can I achieve that? Using `Url.toString` and appending the result manually via `https://foo.bar ++ Url.toString`?

**Comment by bardt** [comment]
Looks like currently there is no other way to achieve this. I'd suggest @nonpop to add `host` and/or`protocol` fields to `Url`, so the type would be something like:

```
type Url a
    = Url (Maybe Protocol, Maybe Host, List (Part a), List ( String, Part a ) )
``` 

Or at least just one `String` to prepend first `/`. The only thing to carefully think about here is that user will probably have his external hostnames as strings like `https://example.com:3000`, but sometimes it would be useful to specify/change different components independently. 

I also need this feature, so I could take and implement it if @nonpop agrees.

**Comment by screendriver** [comment]
That would be great! 👍

**Comment by nonpop** [comment]
Nice to see someone using this and yes, you are welcome to implement it, @bardt! I would prefer the current API remains as it is, so maybe just add a `customRoot` function to be used instead of `root`, and define `root = customRoot ""` or similar?

**Comment by bardt** [comment]
@nonpop I also thought the same way. Will get back with a proposal implementation!

**Comment by nonpop** [comment]
Implemented in 3.0.0.
